base_control/Motor.ino

```diff
@@ -0,0 +1,110 @@
+Motor::Motor() {}
+
+Motor::Motor(byte id,byte d1, byte d2, byte p1, byte e1,byte e2) : _PID(kp, ki, kd) {
+  this->id = id;
+  this->d1 = d1;
+  this->d2 = d2;
+  this->p1 = p1;
+  this->e1 = e1;
+  this->e2 = e2;
+  this->Stop();
+  this->defineOutput();
+  this->changePWM(0);
+
+  //PID
+  _PID.setOutputLimits(0, 255);
+  _PID.setMaxErrorSum(4000);
+  _PID.setSampleTime(TIME_VELOCITY_SAMPLE);
+}
+
+double Motor::getTargetSpeed(){
+  return (moveAll.getTargetTicks()/PULSES_PER_REVOLUTION)*10;

```

Comment:
Pulses per revolution -> kPulsesPerRevolution. Why * 10? where does the 10 come from? Try and not use magic numbers https://help.semmle.com/wiki/display/CCPPOBJ/Magic+numbers

Reply:
Still confused by where you declare the moveAll object.